File: app/src/main/java/com/practice/todo/storage/database/dao/TodoSubItemDao.java
package com.practice.todo.storage.database.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.practice.todo.storage.database.entity.TodoSubItem;

/**
 * TodoSubItem操作类，这里都只需写出定义即可，真正的实现由编译器帮你实现
 */
@Dao
public interface TodoSubItemDao {
    @Query("SELECT * FROM todo_sub_item WHERE parent_id = :parentId")
    TodoSubItem[] loadByParentId(int parentId);

    @Insert
    void insert(TodoSubItem todoSubItem);

    @Update
    void update(TodoSubItem... todoSubItems);

    @Delete
    void delete(TodoSubItem... todoSubItems);
}

File: app/src/main/java/com/practice/todo/storage/database/dao/TodoSubItemDao.kt
package com.practice.todo.storage.database.dao

import androidx.room.*
import com.practice.todo.storage.database.entity.TodoSubItem

@Dao
interface TodoSubItemDao {
    @Query("SELECT * FROM todo_sub_item WHERE parent_id = :parentId")
    fun loadByParentId(parentId: Int): Array<TodoSubItem>

    @Insert
    fun insert(todoSubItem: TodoSubItem)

    @Update
    fun update(vararg todoItems: TodoSubItem)

    @Delete
    fun delete(vararg todoSubItems: TodoSubItem)
}